payment/tests/fun_fake_payment.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+Fake payment page for use in acceptance tests.
+This view is enabled in the URLs by the feature flag `ENABLE_PAYMENT_FAKE`.
+
+Note that you will still need to configure this view as the payment
+processor endpoint in order for the shopping cart to use it:
+
+    settings.CC_PROCESSOR['CyberSource']['PURCHASE_ENDPOINT'] = "/payment/fun_payment_fake"
+
+You can configure the payment to indicate success or failure by sending a PUT
+request to the view with param "success"
+set to "success" or "failure".  The view defaults to payment success.
+"""
+import json
+
+import requests
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic.base import View
+from edxmako.shortcuts import render_to_response
+from opaque_keys.edx.keys import CourseKey
+from student.models import CourseEnrollment
+from courses.models import Course
+
+
+class PaymentFakeView(View):
+    """
+    Fake payment page for use in acceptance tests.
+    """
+
+    # We store the payment status to respond with in a class
+    # variable.  In a multi-process Django app, this wouldn't work,
+    # since processes don't share memory.  Since Lettuce
+    # runs one Django server process, this works for acceptance testing.
+    PAYMENT_STATUS_RESPONSE = "success"
+
+    @csrf_exempt
+    def dispatch(self, *args, **kwargs):
+        """
+        Disable CSRF for these methods.
+        """
+        return super(PaymentFakeView, self).dispatch(*args, **kwargs)
+
+    def post(self, request):
+        """
+        Render a fake payment page.
+
+        This is an HTML form that:
+
+        * Triggers a POST to `postpay_callback()` on submit.
+
+        * Has hidden fields for all the data CyberSource sends to the callback.
+            - Most of this data is duplicated from the request POST params (e.g. `amount`)
+            - Other params contain fake data (always the same user name and address.
+            - Still other params are calculated (signatures)
+
+        * Serves an error page (HTML) with a 200 status code
+          if the signatures are invalid.  This is what CyberSource does.
+
+        Since all the POST requests are triggered by HTML forms, this is
+        equivalent to the CyberSource payment page, even though it's
+        served by the shopping cart app.
+        """
+        # Here we trigger the notification to the payment API (we skipp the ECOMMERCE PART)
+        course_key = request.POST.get('merchant_defined_data1')
+        order_fake = settings.FUN_PAYMENT_FAKE_ORDER(course_key)
```

Comment:
@laurentdavid This `FUN_PAYMENT_FAKE_ORDER` setting is not defined anywhere. This leads me to believe that this PR requires additional changes to be made elsewhere. Is there another PR?

Reply:
Got the information that the following test settings should be added:

```
FEATURES["ENABLE_PAYMENT_FAKE"] = True
CC_PROCESSOR_NAME = 'CyberSource2'
CC_PROCESSOR['CyberSource2']['SECRET_KEY'] = os.environ.get('CYBERSOURCE_SECRET_KEY', '')
CC_PROCESSOR['CyberSource2']['ACCESS_KEY'] = os.environ.get('CYBERSOURCE_ACCESS_KEY', '')
CC_PROCESSOR['CyberSource2']['PROFILE_ID'] = os.environ.get('CYBERSOURCE_PROFILE_ID', '')
CC_PROCESSOR['CyberSource2']['PURCHASE_ENDPOINT'] = '/payment/fun_payment_fake'

FUN_PAYMENT_FAKE_ORDER = lambda course_key: {
    'number': 'FUN-100056',
    'total_excl_tax': '100.00',
    'date_placed': '2016-03-17T10:07:36Z',
    'lines': [
        {
            'product': {
                'attribute_values': [
                    {
                        'name': 'course_key',
                        'value': course_key,
                    }
                ]
            }
        }
    ]
}
COMMON_TEST_DATA_ROOT = COMMON_ROOT / "test" / "data"
```